portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -2771,6 +2778,92 @@ private long _getValidFileEntryTypeId(
 		}
 	}
 
+	private void _inheritRolesPermissions(
+			long propagationParentFolderId, DLFileEntry dlFileEntry)
+		throws PortalException {
+
+		Set<String> commonPermissions = new HashSet<>(
+			Arrays.asList(
+				ActionKeys.DELETE, ActionKeys.PERMISSIONS, ActionKeys.UPDATE,
+				ActionKeys.VIEW));
+
+		long companyId = dlFileEntry.getCompanyId();
+		String parentClassName = DLFolderConstants.getClassName();
+
+		Set<String> dlFolderAllActionIds = SetUtil.fromCollection(
+			ResourceActionsUtil.getModelResourceActions(
+				DLFolderConstants.getClassName()));
```

Comment:
Why do we have to use Set here? I don't think there is any case that the list includes the duplicate element.

Reply:
Should be `dlFolderModelActionIds` instead

Reply:
I use this following with Loc's last logic https://github.com/tototrinh/liferay-portal/commit/46cbc2ba57073539e441d8730d6887e6f2ccd0e8#diff-7d374c46b1d3f63b8fb1a08828d641e05589d5d05d687ddf159ae0e05062f7deR1480 for synonymous. 
I renamed it. Thanks

Reply:
If it's not necessary to use the set here and we missed it in the previous review, we can also refactor Loc's code.

Reply:
Moved it inline so we don't have to define Set here.

Reply:
Fixed it. Thanks